thunder/core/transforms.py

```diff
@@ -1440,16 +1440,24 @@ def grad_func(*args, **kwargs):
         return grad_func
 
     class _GradTransform(Transform):
-        def transform_trace_additionally(self, trc: Trace, *, executors_list: Sequence[Any]) -> Trace:
+        def transform_traces_pre_prologue(
+            self,
+            prologue_trc: Trace,
+            computation_trc: Trace,
+            epilogue_trc: Trace | None,
+            *,
+            executors_list: Sequence[Any],
+        ) -> Trace:
```

Comment:
```suggestion
        ) -> Trace:
            computation_trace = dce(computation_trace)
```
As discussed on slack, but here for transparency: we need a dce pass here...